### PR TITLE
Refactor zephyr timing implementation into two separate files

### PR DESCRIPTION
**Pull request by erlingrj**
Clean up Zephyr platform implementation:
- Remove the use of preprocessor to conditionally compile the two different timing APIs
- Implement the kernel timing API and counter timing API in two separate files
- remove the `_lf_` prepending of all the static local variables. They are static and local to that file, so no need to add long complicated variable names that are hard to parse 

Closes https://github.com/lf-lang/lingua-franca/issues/2063

### CODE REVIEW

**Review by lhstrh — APPROVED** [review]
Nice! Looks great to me. I guess this doesn't require any changes in the code generator?

**Comment by erlingrj** [comment]
Correct, no changes required in `lingua franca`
